### ISSUE TITLE
gnome terminal size(height) shrank while focus changed

### ISSUE THREAD

**Issue by zigzed** [snwh/paper-gtk-theme#66]
version: git commit: a47a92b36dc957f58a174462e30c0988b8e946d7 (existed in early versions too)
reproduce:
1. open application 'gnome-terminal' with size 80 x 24.
2. open a new terminal tab. ( press CTRL+SHIFT+T )
3. change focus to another application ( press ALT+TAB )
4. change focus back to gnome-terminal, and the height is shrank (press ALT+TAB again)
5. repeat step 3, 4

**Comment by truongan** [comment]
Confirm this. It's a killer bug that prevent me from using this theme.

**Comment by axel-sommerfeldt** [comment]
I cannot confirm this issue, no problem here. (GNOME Shell 3.14.3, Fedora 21 up-to-date, tested with a47a92b and eefb7ea )

**Comment by bartlibert** [comment]
I can confirm this (Gnome 3.14.1, Debian Jessie/sid), but it only happens when I have tabs in my terminal, without tabs it's fine.

**Comment by truongan** [comment]
@bartlibert :+1: 
I confirm this bug and @bartlibert's work around. Perhaps @axel-sommerfeldt failed to repoduce the problem he missed step 2 (create new terminal tab)

**Comment by axel-sommerfeldt** [comment]
No, I haven't missed that step, I have followed the instructions exactly. I switch between Firefox and the Terminal (w/ two tabs) with ALT-TAB and nothing bad happens. I open the gnome calculator and switch between it and the terminal, and nothing bad happens. I updated Paper to version 62efe41 and nothing bad happens. (I have setup Paper as "Windows", "GTK+", and "Shell Theme". "Symbols" are on "Gnome" and "Mouse Pointer" on "Adwaita".) Maybe this is a problem with Gnome versions prior 3.14.3?

**Comment by truongan** [comment]
@axel-sommerfeldt : Well I'm on Gnome 3.14.2 so I guess this is indead a  problem with Gnome version.

However, I tried 49f7cfe9ab today and everything seem fine. So this issuse may be consider closed

**Comment by truongan** [comment]
I'm afraid I have to stir this issues up. I'm on Gnome 3.16.1, Archlinux and this bug still persist. So I guess it's not a problem with gnome version.
The work around provided by @bartlibert  is still effective though

**Comment by snwh** [comment]
There's still mixed confirmations and I've not been able to confirm it either.

**Comment by golimpio** [comment]
I can confirm this issue...I'm running Ubuntu 15.04 and I was running the latest Paper GTK theme from trunk:
- Using the default Ubuntu Ambiance theme
- Opened a new terminal (Ctrl-Alt-T)
- Created a new tab (Ctrl-Shit-T)
- Changed the theme to Paper GTK
- Click on the terminal again
- Changed tabs
- The terminal window just shrank

Thanks!

**Comment by clauseggers** [comment]
I have had the same issue periodically. Changing from the terminal to another app and then back to the terminal decrease the hight of the terminal on every such change of apps.

**Comment by aronbohl** [comment]
I'm on a new install of Ubuntu 15.04 (Unity) and can reproduce this every time with the following steps.

1) enable Paper theme
2) open gnome terminal
3) open new tab in the terminal
4) window shrinks

Also, I can change the size back of the window back to normal but then when the terminal loses focus to another application, the window shrinks again.

I've gotten this to happen on multiple machines.

**Comment by snwh** [comment]
I'm not sure what's going on there. Does this same issue occur in other themes?

**Comment by clauseggers** [comment]
It only happens in Paper AFAICS, but I have only tested Adwaita, Numix, and HighContrast and they did not do it.

**Comment by bartlibert** [comment]
I have tested some other themes (some of the Zuki* themes a.o.) and have not seen it in any theme except Paper.

**Comment by webdv** [comment]
Facing the same issue, using debian 8, with paper theme only, occurs only when opened another tab, any solutions?

**Comment by cholmcc** [comment]
Hi,

I'm on Debian "sid" (just upgraded) using GNOME 3.16.2 (at least that's what the system settings says), with a fresh install of the Paper theme, and I have the same problem, The solution of @bartlibert works though. 

I used to have the problem on another computer (not so much up-to-date), but not on this one, but now  I see it on this one, but not the other one - strange.
.

**Comment by cavemandaveman** [comment]
I had this same issue on Gentoo, but can confirm it is fixed after pulling down the most recent paper-gtk-theme source and installing over the previous version. 

Not sure, but @bartlibert workaround might have been the solution because the latest code does not even include the "border: none;" line in Paper/gtk-3.0/widgets/notebook.css

**Comment by cholmcc** [comment]
I see the same issue with paper theme 

  2.1+r265~daily~ubuntu16.04.1

on Debian "sid" using Gnome 3.20.2

In fact, every time `RETURN` is pressed, the terminal window shrinks by one line. 

There's no "border: 0" line to remove in `notebook.css` so that isn't the problem.

**Comment by zachtib** [comment]
I just noticed this issue on my computer as well. Arch Linux, Gnome 3.22, Wayland session on amdgpu drivers

**Comment by AlexanderMB** [comment]
Arch Linux, Gnome 3.22, Wayland, Intel: gnome-terminal shrinks after changing focus away when using Paper theme. Does not happen when using Adwaita theme.

Arch Linux, Gnome 3.22, X11, nVidia: bug does not happen with Paper or Adwaita

**Comment by auserdude** [comment]
Arch Linux, Gnome 3.22, same issue here.

**Comment by tarmtrollet** [comment]
same here with fedora 25 and gnome 3.22. but for me it shrinks even if no tabs are open.

**Comment by shawniac** [comment]
Confirming this issue on Archlinux with Kernel 4.11.3 and Gnome 3.24.2 on Wayland.
I'm using tmux, thus I don't have multiple gnome-tabs but tmux "tabs", in every gnome-terminal and upon loosing and regaining focus the terminal decreases it's height by one line.

I just tested a pure gnome-terminal with no extra tabs or tmux, and this bug is still present! (This means the tab theory doesn't work for me)

If I might add, this bug almost drove me mad, because I thought there can't possibly be a bug that would cause this - turns out I was wrong.

Luckily I found this issue thread with some google fu! 😛

**Comment by willcarroll7** [comment]
I also have the issue currently on **Fedora 25** with **Kernel 4.11.9** running **Gnome 3.22.3** on **Wayland**. Just as @shawniac has I don't need any tabs to be open for the terminal to resize. It shrinks by one line every time I switch focus back to the terminal (via alt + tab or other).

**Comment by LBreda** [comment]
Same here with Debian Buster, Gnome 3.22.2, Wayland.

No need to have multiple tabs open.

**Comment by pziarsolo** [comment]
Same here with Debian Buster/sid. Gnome 3.26, Wayland and version 2.1+r265-daily-ubuntu17.04.1.

No problem with X11 nor adwaita theme.

**Comment by nd2s** [comment]
Having this problem as well on Arch Linux `4.13.11-1-ARCH`, Wayland, Gnome 3.26.2.
Serious bug that prevents me from using this theme.

Has this really been an issue for years?

**Comment by kappapiana** [comment]
Have the same issue myself. Ubuntu 17.10 and 17.04 (same behaviour, two different machines). Switching to another theme, the behaviour is not there any more

**Comment by f4ww4z** [comment]
Same issue with @shawniac. On default Ubuntu 17.10 (Gnome 3.24).

**Comment by rhaces** [comment]
Yep, still an issue with Ubuntu 17.10... Unfortunately the line that is mentioned above to remove is no longer in the file so I'm not sure how to work around this.... Great theme but this is a deal breaker so I will have to look for a new one.

**Comment by timothee-haudebourg** [comment]
Same problem here with Arch Linux + Wayland + GNOME 3.26.2. This is really annoying...

**Comment by jagibson** [comment]
Same problem.  Ubuntu 17.10, wayland (intel graphics).  happens with or without tabs.

**Comment by hirnschallsebastian** [comment]
Same problem on Ubuntu 17.10 wayland (nvidia graphics). No issue with animations turned off in the tweak tool.
